From 481ec9b7eb87b55418269c285f53f490d805b752 Mon Sep 17 00:00:00 2001
From: Chih-Hung Hsieh <chh@google.com>
Date: Tue, 9 Dec 2014 14:28:07 -0800
Subject: [PATCH 60/95] Fix wrong int argument of FOURCC.

Clang compiler reports error for aosp_x86-eng target:

.../MPEG4Extractor.cpp:1538:14: error: case value is not a constant expression
case FOURCC('\xA9', 'x', 'y', 'z'):
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
..../MPEG4Extractor.cpp:1538:14: note: left shift of negative value -87

Change-Id: I3ef1aa710cd1ce84ca46cdfbe37876bb9a1140bd
---
 media/libstagefright/MPEG4Extractor.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/media/libstagefright/MPEG4Extractor.cpp b/media/libstagefright/MPEG4Extractor.cpp
index ec1b104..a75da2a 100644
--- a/media/libstagefright/MPEG4Extractor.cpp
+++ b/media/libstagefright/MPEG4Extractor.cpp
@@ -1552,13 +1552,13 @@ status_t MPEG4Extractor::parseChunk(off64_t *offset, int depth) {
             break;
         }
 
-        // @xyz
-        case FOURCC('\xA9', 'x', 'y', 'z'):
+        // ©xyz
+        case FOURCC(0xA9, 'x', 'y', 'z'):
         {
             *offset += chunk_size;
 
-            // Best case the total data length inside "@xyz" box
-            // would be 8, for instance "@xyz" + "\x00\x04\x15\xc7" + "0+0/",
+            // Best case the total data length inside "©xyz" box
+            // would be 8, for instance "©xyz" + "\x00\x04\x15\xc7" + "0+0/",
             // where "\x00\x04" is the text string length with value = 4,
             // "\0x15\xc7" is the language code = en, and "0+0" is a
             // location (string) value with longitude = 0 and latitude = 0.
-- 
2.5.0

